framework/src/multiapps/MultiApp.C

```diff
@@ -1093,16 +1093,17 @@ rankConfig(dof_id_type rank,
   // A "slot" is a group of procs/ranks that are grouped together to run a
   // single (sub)app/sim in parallel.
 
-  auto slot_size = std::max(std::min(nprocs / napps, max_app_procs), min_app_procs);
-  dof_id_type nslots = std::min(nprocs / slot_size, napps);
-  auto leftover_procs = nprocs - nslots * slot_size;
-  auto apps_per_slot = napps / nslots;
-  auto leftover_apps = napps % nslots;
+  dof_id_type slot_size =
+      std::max(std::min(nprocs / napps, (dof_id_type)max_app_procs), (dof_id_type)min_app_procs);
+  processor_id_type nslots = std::min(nprocs / slot_size, napps);
+  processor_id_type leftover_procs = nprocs - nslots * slot_size;
+  dof_id_type apps_per_slot = napps / nslots;
+  dof_id_type leftover_apps = napps % nslots;
```

Comment:
```suggestion
  processor_id_type slot_size =
      std::max(std::min(nprocs / napps, (dof_id_type)max_app_procs), (dof_id_type)min_app_procs);
  processor_id_type nslots = std::min(nprocs / slot_size, napps);
  processor_id_type leftover_procs = nprocs - nslots * slot_size;
  processor_id_type apps_per_slot = napps / nslots;
  processor_id_type leftover_apps = napps % nslots;
```

Reply:
I think everything connected to apps should stay `dof_id_type`. I think stochastic tools started out from the need of statistics from 10s of billions of simulations without file io which slows it down. In certain cases the number of apps can be the same as the number of simulations needed. For this reason, I'd like to keep those types.

Reply:
ok yea that makes sense